.github/workflows/CI.yml

```diff
@@ -45,29 +47,57 @@ jobs:
 
       # TODO: We shouldn't require separate caches per Julia version but allow cache restores to work
       # https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
-      - uses: actions/cache/restore@v3
-        id: bazel-cache
+      - name: Restore build cache
+        uses: actions/cache/restore@v3
+        id: build-cache
         with:
-          key: bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.hash-${{ hashFiles('deps/WORKSPACE.bazel.tpl', 'deps/BUILD.bazel') }}
-          path: ~/.cache/bazel
+          key: build-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.hash-${{ hashFiles('deps/WORKSPACE.bazel.tpl', 'deps/BUILD.bazel') }}
+          path: ~/.cache
           restore-keys: |
-            bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.
-            bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.
-            bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.
+            build-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.
+            build-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.
+
+      # Based upon:
+      # https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#building-ray-on-linux-macos-full
+      - name: Build Ray CLI
+        run: |
+          pip install --upgrade pip wheel
+          pushd deps/ray
+
+          # The Ray BUILD.bazel includes a bunch of `copy_to_workspace` rules which copy build output
+          # into the Ray worktree. When we only restore the Bazel cache then re-building causes these
+          # rules to be skipped resulting in `error: [Errno 2] No such file or directory`. By manually
+          # saving/restoring these files we can work around this.
+          RAY_GEN_CACHE_DIR=~/.cache/ray-generated
+          if [ -d "$RAY_GEN_CACHE_DIR" ]; then
+              dest=$(pwd)
+              pushd $RAY_GEN_CACHE_DIR
+              cp -rp --parents \
```

Comment:
`--parents` only seems to be available in linux - I guess this precludes us testing the build on macos?

Reply:
We can update this in the future to use multiple steps. We would just need to make the directory first and then perform the copies. Using `--parents` should be faster as we can avoid having to perform multiple copy commands. Most likely we'd just have an alternate code path for macOS